paddle/fluid/framework/ngraph_operator.h

```diff
@@ -0,0 +1,161 @@
+/* Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+
+#ifdef PADDLE_WITH_NGRAPH
+
+#include <algorithm>
+#include <atomic>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+#include "glog/logging.h"  // For VLOG
+#include "paddle/fluid/framework/attribute.h"
+#include "paddle/fluid/framework/framework.pb.h"
+#include "paddle/fluid/framework/lod_tensor.h"
+#include "paddle/fluid/framework/ngraph_bridge.h"
+#include "paddle/fluid/framework/op_info.h"
+#include "paddle/fluid/framework/op_kernel_type.h"
+#include "paddle/fluid/framework/operator.h"
+#include "paddle/fluid/framework/program_desc.h"
+#include "paddle/fluid/framework/scope.h"
+#include "paddle/fluid/framework/selected_rows.h"
+#include "paddle/fluid/framework/tensor.h"
+#include "paddle/fluid/platform/device_context.h"
+#include "paddle/fluid/platform/variant.h"
+
+#include "ngraph/ngraph.hpp"
+
+namespace paddle {
+namespace framework {
+
+class FusedOperator : public OperatorBase {
+ public:
+  explicit FusedOperator(
+      const ProgramDesc& prog, size_t block_id,
+      std::vector<std::unique_ptr<OperatorBase>>& ops,  // NOLINT
+      const std::string& type = "", const VariableNameMap& inputs = {},
+      const VariableNameMap& outputs = {}, const AttributeMap& attrs = {})
+      : OperatorBase(type, inputs, outputs, attrs),
+        pdesc(prog),
+        block(block_id) {
+    for (auto& op : ops) {
+      fused_ops.push_back(std::move(op));
+    }
+
+    process();
+  }
+
+  void RunImpl(const Scope& scope, const platform::Place& place) const final;
+
+ private:
+  const ProgramDesc pdesc;
+  size_t block;
+  std::vector<std::shared_ptr<OperatorBase>> fused_ops;
+  std::unordered_map<std::string, proto::VarType::Type> var_type_map;
+  std::unordered_set<std::string> persistables;
+
+  // op idx in fused_ops
+  std::vector<int> pre_feed_idx{0, -1};
+  std::vector<int> post_fetch_idx{0, -1};
+  std::vector<int> ng_op_idx{0, -1};
+  std::unordered_set<std::string> fetches;
+  // 0 = default; 1 = (is_test && not is_complete)
+  // 2 = (is_test && is_complete)
+  // 3 = (is_training && not is_complete)
+  // 4 = (is_training && is_complete)
+  // int is_test_or_train = 0;
+
+  void process();
+};
+
+class NgraphOperator {
```

Comment:
It seems this is not used in other places, can you declare it in .cc file?

Reply:
Let me look into this.